src/main/java/core/basesyntax/FruitShop.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import core.basesyntax.dto.Transaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import core.basesyntax.service.FileReaderImpl;
+import core.basesyntax.service.FileWriter;
+import core.basesyntax.service.FileWriterImpl;
+import core.basesyntax.service.FruitParser;
+import core.basesyntax.service.ReportFruitImpl;
+import core.basesyntax.service.ValidatorImpl;
+import core.basesyntax.strategy.BalanceOperation;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.PurchaseOperation;
+import core.basesyntax.strategy.SupplyOperation;
+
+public class FruitShop {
+    public static void main(String[] args) {
+        Map<String, Operation> operationsMap = new HashMap<>();
+        operationsMap.put("b", new BalanceOperation());
+        operationsMap.put("s", new SupplyOperation());
+        operationsMap.put("p", new PurchaseOperation());
+        operationsMap.put("r", new SupplyOperation());
+        FileReaderImpl fileReader = new FileReaderImpl();
+        List<String> listData = fileReader.readFromFile("src/main/java/core/basesyntax/resources/input.csv");
+        listData.remove(0);
+        ValidatorImpl validator = new ValidatorImpl();
```

Comment:
not sure abou this so I only ask you but you can do as you think.
Maybe it will be better to check our data with Validator in FruitParser class. So in our main class will be less code and check if data is valid will be near parse logic.

Reply:
To my mind it won't be right according to single redponsibility rule (in FruitParser data parsed, and in Validator we check it)